### ISSUE TITLE
Accessibility issue in Stack view - the block section is not associated with it's corresponding label

### ISSUE THREAD

**Issue by TeodoraNozharova** [vmware/clarity#5609]
### Describe the bug

Screen readers cannot associate the block section of the Stack view with it's corresponding label.
Currently there is a `div` wrapper element that has a `region` role assigned to it, but it's lacking the `aria-labelledby` attribute.

### How to reproduce
Reproduced with the basic use of accordion, described in V4 documentation:
https://stackblitz.com/edit/clarity-v4-light-theme-stack-view?file=src/app/app.component.html

### Steps to reproduce the behavior:

Read the elements of any Stack View example using a screen reader
Observe that the screen reader does not announce any information regarding how the content of the Stack view is corresponding with the label, where the region starts or ends.

### Expected behavior
According to [WAI ARIA practices](https://www.w3.org/TR/wai-aria-practices-1.1/examples/accordion/accordion.html) each `region` should have assigned `aria-labelledby` that references the header button that expands and collapses the region. This way the panel can be clearly identified by a screen reader as a region that corresponds to a given label.

**Comment by Jinnie** [comment]
CC @scroniser Santina, can we please verify if we track this internally?
Thanks!

**Comment by scroniser** [comment]
@TeodoraNozharova, thank you for reporting this issue. 

@Jinnie, confirmed that this is in our [internal tracking system](https://jira.eng.vmware.com/browse/VPAT-608).

**Comment by TeodoraNozharova** [comment]
@scroniser, @Jinnie thanks for looking into this!

@scroniser the linked jira bug is marked as resolved, but the [fix](https://github.com/vmware/clarity/commit/1280cf1e936705b4935bafef983e32561fbadf94) does not include adding `aria-labelledby` , only `region` as well as additional fixes such as removing the description list structure.

So in the current state the screen readers are still not able to interpret the stack view structure properly. Could this issue be considered separately?

**Comment by Jinnie** [comment]
I'll close this issue here, so we can avoid double-tracking. If needed to have additional bug for the `aria-labelledby` this can be done directly with the A11Y team.

**Comment by scroniser** [comment]
@TeodoraNozharova, Just wanted to follow up, thank you for the clarification. It looks like I pasted in the wrong bug -- we have a separate bug logged here: [VPAT-592](https://jira.eng.vmware.com/browse/VPAT-592).

**Comment by TeodoraNozharova** [comment]
@scroniser Thanks, this issue seems more related, per your latest comment in the issue.
